app/src/main/java/de/danoeh/antennapod/fragment/AllEpisodesFragment.java

```diff
@@ -92,13 +93,18 @@ protected void updateFilter(Set<String> filterValues) {
     @NonNull
     @Override
     protected List<FeedItem> loadData() {
-        return feedItemFilter.filter(DBReader.getRecentlyPublishedEpisodes(0, page * EPISODES_PER_PAGE));
+        return loadDataFromDB(0, page * EPISODES_PER_PAGE);
     }
 
     @NonNull
     @Override
     protected List<FeedItem> loadMoreData() {
-        return feedItemFilter.filter(DBReader.getRecentlyPublishedEpisodes((page - 1) * EPISODES_PER_PAGE,
-                EPISODES_PER_PAGE));
+        return loadDataFromDB((page - 1) * EPISODES_PER_PAGE, EPISODES_PER_PAGE);
+    }
+
+    private List<FeedItem> loadDataFromDB(int offset, int limit) {
+        LongList loadedQueueIds = DBReader.getQueueIDList();
+        List<FeedItem> loadedFeedItems = DBReader.getRecentlyPublishedEpisodes(offset,limit);
+        return feedItemFilter.filter(loadedFeedItems,loadedQueueIds);
```

Comment:
There is currently an open PR (#4563) that changes something here, too. This will likely cause conflicts. To be honest, I think it would be better to do the changes a bit more gradually in multiple PRs.

Reply:
OK, noted for future.